lib/ae_mdw/db/name.ex

```diff
@@ -37,6 +42,78 @@ defmodule AeMdw.Db.Name do
     auction_end + :aec_governance.name_claim_max_expiration(proto_vsn(auction_end))
   end
 
+  @spec expirations_mutation(Blocks.height()) :: NamesExpirationMutation.t()
+  def expirations_mutation(height) do
+    name_mspec =
+      Ex2ms.fun do
+        Model.expiration(index: {^height, name}) -> name
+      end
+
+    auction_mspec =
+      Ex2ms.fun do
+        {:expiration, {^height, name}, tm} -> {name, tm}
+      end
+
+    expired_names = :mnesia.dirty_select(Model.ActiveNameExpiration, name_mspec)
+    expired_auctions = :mnesia.dirty_select(Model.AuctionExpiration, auction_mspec)
+
+    NamesExpirationMutation.new(height, expired_names, expired_auctions)
+  end
+
+  @spec expire_name(Blocks.height(), Names.plain_name()) :: :ok
+  def expire_name(height, plain_name) do
+    m_name = cache_through_read!(Model.ActiveName, plain_name)
+    m_exp = Model.expiration(index: {height, plain_name})
+
+    owner = Model.name(m_name, :owner)
+    cache_through_write(Model.InactiveName, m_name)
+    cache_through_write(Model.InactiveNameExpiration, m_exp)
+    cache_through_delete(Model.ActiveName, plain_name)
+    cache_through_delete(Model.ActiveNameOwner, {owner, plain_name})
+    cache_through_delete(Model.ActiveNameExpiration, {height, plain_name})
+    Ets.inc(:stat_sync_cache, :inactive_names)
+    Ets.dec(:stat_sync_cache, :active_names)
+    Log.info("[#{height}] expiring name #{plain_name}")
+  end
+
+  @spec expire_auction(Blocks.height(), Names.plain_name(), Names.auction_timeout()) :: :ok
+  def expire_auction(height, plain_name, timeout) do
+    {_, {_, txi}, _, owner, bids} =
+      bid_key = ok!(cache_through_prev(Model.AuctionBid, bid_top_key(plain_name)))
+
+    previous = ok_nil(cache_through_read(Model.InactiveName, plain_name))
+    expire = expire_after(height)
+
+    m_name =
+      Model.name(
+        index: plain_name,
+        active: height,
+        expire: expire,
+        claims: bids,
+        auction_timeout: timeout,
+        owner: owner,
+        previous: previous
+      )
+
+    m_name_exp = Model.expiration(index: {expire, plain_name})
+    m_owner = Model.owner(index: {owner, plain_name})
+
+    ensure_no_exsiting_active_name_expiration(plain_name)
```

Comment:
if not removed, can also be moved to outside the transaction block.

Reply:
@jyeshe definitely. The idea is to move all of the mutations into separate mutation modules. The next step would be to trim those mutations and move as much code outside of them as possible